src/entities/applet/model/hooks/useStartEntity.ts

```diff
@@ -1,19 +1,33 @@
 import { useQueryClient } from '@tanstack/react-query';
 
-import { FlowProgress, StoreProgressPayload } from '@app/abstract/lib';
+import {
+  FlowProgress,
+  LookupMediaInput,
+  StoreProgressPayload,
+} from '@app/abstract/lib';
 import { ActivityFlowRecordDto, AppletDetailsResponse } from '@app/shared/api';
 import {
   getAppletDetailsKey,
   getDataFromQuery,
+  isAppOnline,
   useAppDispatch,
   useAppSelector,
 } from '@shared/lib';
 
-import { onBeforeStartingActivity } from '../../lib';
+import { onBeforeStartingActivity, onMediaReferencesFound } from '../../lib';
 import { selectInProgressApplets } from '../selectors';
 import { actions } from '../slice';
 
-function useStartEntity() {
+type StartResult = {
+  startedFromScratch?: boolean;
+  cannotBeStartedDueToMediaFound?: boolean;
+};
+
+type UseStartEntityInput = {
+  lookupMedia: (input: LookupMediaInput) => boolean;
```

Comment:
The function returns a boolean, hence it should have an appropriate name, like `hasMedia` or `hasMediaContent`